crates/dbsp/src/operator/filter_map.rs

```diff
@@ -123,7 +123,7 @@ pub trait FilterMap<C> {
     /// IntoIterator`, e.g., `Option<>` or `Vec<>`.
     fn flat_map<F, I>(&self, func: F) -> Stream<C, OrdZSet<I::Item, Self::R>>
     where
-        F: Fn(Self::ItemRef<'_>) -> I + 'static,
+        F: FnMut(Self::ItemRef<'_>) -> I + 'static,
```

Comment:
what's the use case for this?

Reply:
It allows buffer reuse, the jit generates flat_map calls that reuse vectors